src/lib/DataConnectStore.js

```diff
@@ -139,9 +139,16 @@ export default class DataConnectStore {
   }
 
   updateAccount (connectorId, accountIdx, values) {
-    let connector = this.find(c => c.id === connectorId)
-    connector.accounts[accountIdx] = values
-    return this.putConnector(connector)
+    let connector = this.findConnectorById(connectorId)
```

Comment:
To avoid to find the connector from a given id, we should directly pass it as parameter : 
```js
updateAccount(konnector, account, values) {
```
As it is done for connectAccount()
To do so, we may pass directly the expected values into `ConnectManagement` component, as it is already done for `connectAccount` here https://github.com/cozy/cozy-data-connect/blob/master/src/containers/ConnectorManagement.jsx#L159